receiver/githubreceiver/traces_receiver.go

```diff
@@ -0,0 +1,132 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package githubreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/githubreceiver"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/google/go-github/v67/github"
+	"github.com/julienschmidt/httprouter"
```

Comment:
The main reason is that it is already used in the WebHook receiver which this skeleton resembles closely. It's "supposed" to be more performant according to their docs, and a few well known Go frameworks (like Gin) use it under the hood. I do have a working implementation using native `net/http` but I wanted to keep things closely resembling the tried and true webhook event receiver.

That being said, when I went back to look at this library, I noticed that, while actively being developed, it hasn't had a release since 2019. I'd be happy to use native net/http if that's the recommendation here.

Reply:
@MovieStoreGuy  - upon further investigation, I've opted to remove `julienschmidt/httprouter` in favor of `gorilla/mux`. It provides a similarly simple interface for creating routes, is better maintained, and is used more widely across components within contrib. From this pull requests' perspective, it does feel like overkill. However, in subsequent PRs, I think the added functionality for routing through gorilla/mux will be a little simpler. I'll be adding at least 2 more, potentially 3, routes within this collector in the end state. 

If you feel that I haven't addressed this feedback, feel free to re-open. 😊